config/swarm.go

```diff
@@ -137,10 +139,8 @@ type ConnMgr struct {
 // <https://github.com/libp2p/go-libp2p-resource-manager#readme>
 type ResourceMgr struct {
 	// Enables the Network Resource Manager feature
-	Enabled Flag `json:",omitempty"`
-
-	/* TODO: decide if and how we want to expose limits in our config
-	Limits *ResourceMgrScopeConfig `json:",omitempty"` */
+	Enabled Flag                      `json:",omitempty"`
+	Limits  *rcmgr.BasicLimiterConfig `json:",omitempty"`
```

Comment:
Ok to reuse type from `go-libp2p-resource-manager`, but if they change the struct then our users may get panics.

One way to guard against that is to add  these structs to `checkImplicitDefaults`  in `rcmgr_defaults.go` – mind adding it there (if there is no better way)?

Reply:
What kind of change are you talking about?

Reply:
Hm.. ok, i was bit too paranoid. If the next version of go-libp2p renames `BasicLimiterConfig.System` to `BasicLimiterConfig.Foo` then we will get compile errors in `core/node/libp2p/rcmgr.go`, so it won't be a silent breakage. 

I guess if that happens, we either ask go-libp2p to undo the change, or write config migration for limits set by our users in `Swarm.ResourceMgr.Limits.System`.